### DISCUSSION TITLE
Smart-Meter for testing environment

### DISCUSSION THREAD

**Discussion by e-swissolar** [scs/smartmeter-datacollector#24] [Q&A] [answered]
Hello there, I am working on a blockchain application and intent to use the smartmeter-datacollector. I am looking for a smart-meter (not industrial) for development and testing. What are the requirements for a compatible smart-meter. Could you recommend me one? Thanks and regards

**Comment by raymar9** [comment, marked as answer]
Hi
At the moment, we developed and tested the smartmeter-datacollector for the smart meters Landis+Gyr E450 (over wired M-Bus) and Iskraemeco AM550 (over P1 DSMR interface). See our [Wiki](https://github.com/scs/smartmeter-datacollector/wiki). The apps software architecture is defined to be extensible for further meter types in the future - however some development is required, obviously.

Just out of curiosity, how do you plan the blockchain integration? Do you implement a new "sink" ([see SW architecture](https://github.com/scs/smartmeter-datacollector/wiki/Architecture-&-Design)) or do you use the MQTT sink and subscribe the blockchain service to the MQTT broker?
Best regards

**Reply by e-swissolar** [reply]
Thanks, so if for example, I would take such a smart meter - >[ link](https://new.abb.com/low-voltage/products/system-pro-m/energy-efficiency-devices/energy-meters) with m-bus, would this work out?

I still have to understand your solution. There are different ways to inject data in to a DLT. It depends on the use-case. For storing readings on-chain the MQTT broker subscription is an option.

**Reply by raymar9** [reply]
If you have an energy meter like one from your link, you still need to extend the current state of our project with another smart meter type, therefore write some code. I highly assume that you cannot just reuse the L+G E450 meter type for your meter.
However, I could not find any datasheet or manual with explanations about the M-Bus interface. Can you tell me the exact meter model that you want to buy and/or the M-Bus interface specification?

**Reply by e-swissolar** [reply]
I am not really sure which one to buy. The ABB is one from many I could find online. I think it has to be an MID with m-bus interface and not so expensive.  When you go on the chapter 10 of the [link](https://search.abb.com/library/Download.aspx?DocumentID=2CMC484002M0201&LanguageCode=en&DocumentPartId=&Action=Launch) you can read the description. It would be really helpful to work for an implementation for a testing environment. Please let me know how we could proceed and how I could contribute. Many thanks

**Reply by raymar9** [reply]
After a quick look at the manual I can confirm that the communication over M-Bus works differently for this meter in comparison to the already implemented (L+G & Iskra). M-Bus only covers the physical- and some of the data-link layer. How the data is transmitted on higher layers is another topic. Our software expects the smart meters to send data automatically every x seconds without being requested for. The protocols used are M-Bus (physical), HDLC (data-link) and DLMS as data messaging protocol.

The ABB meter uses the same OBIS codes (manufacturer independent identification of value types like "active power on phase 1"), however I do not assume that HDLC and DLMS are used protocols with this meter. I found the standards EN 13757-2 and EN 13757-3 in the manual which do not refer to HDLC or DLMS.

Therefore I guess that probably this is not a suitable project as base for your intention since only little code of this project could be reused. Our project focuses on smart meters which are installed in buildings by electricity suppliers.

**Reply by e-swissolar** [reply]
Thank you very much for your response. You already helped me a lot. Your project is a very useful initiative, and I am aware about your focus. Regarding my project, maybe I have to rethink the set-up of our experiment. We do not have to reinvent the wheel, but nevertheless it makes sense to check possible transmission protocols for such not industrial meters. Hopefully we will have a standard for Switzerland soon.
